### PR TITLE
Add -WithSource to bicep publish

### PR DESCRIPTION
**Pull request by StephenWeatherford**
## Description

Adds a new switch to Publish-AzBicepModule called "-WithSource" which causes "--with-source" to be passed to the Bicep CLI (part of https://github.com/Azure/bicep/issues/11435).

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by StephenWeatherford** [comment]
Ready for review.

**Review by anthony-c-martin — APPROVED** [review]

**Review by shenglol — APPROVED** [review]

**Comment by StephenWeatherford** [comment]
Added tests

**Comment by shenglol** [comment]
Hi @isra-fel, could you help review the PR when you get a chance?

File: src/Resources/Resources/ChangeLog.md [hunk 1]
```diff
@@ -27,6 +27,7 @@
 * Fixed reporting duplicate warnings when compiling Bicep files.
 * Updated New and Set Management Group cmdlets to allow DeploymentSubscription to be optional.
 * Fixed inexplicable error message when subscription and scope are neither provided in RoleAssignment/RoleDefinition related commands. [#22716]
+* Added `-WithSource` parameter to `Publish-AzBicepModule` for publishing source with a module (currently experimental)
```
Comment (dolauli) [inline src/Resources/Resources/ChangeLog.md:30]: Should be under Upcoming Release?

**Review by dolauli — CHANGES_REQUESTED** [review]

File: src/Resources/ResourceManager/ResourceManager.sln [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 17
+VisualStudioVersion = 17.5.002.0
+MinimumVisualStudioVersion = 10.0.40219.1
```
Comment (BethanyZhou) [inline src/Resources/ResourceManager/ResourceManager.sln:5]: Please remove this file

File: src/Resources/Resources.Test/SessionRecords/Microsoft.Azure.Commands.Resources.Test.ScenarioTests.DeploymentTests/TestNewDeploymentTemplateSpec.json [hunk 3]
```diff
@@ -1,4 +1,4 @@
-{
+  {
```
Comment (BethanyZhou) [inline src/Resources/Resources.Test/SessionRecords/Microsoft.Azure.Commands.Resources.Test.ScenarioTests.DeploymentTests/TestNewDeploymentTemplateSpec.json:1]: Please revert this change as no scenario test is recorded.

File: src/Resources/Resources/ChangeLog.md [hunk 4]
```diff
@@ -21,6 +21,7 @@
 ## Upcoming Release
 * Used utf8 encoding for reading stdout & stderr when invoking Bicep. [#23246]
 * Fixed regression in `Publish-AzBicepModule` [Azure/bicep/12461](https://github.com/Azure/bicep/issues/12461)
+* Added `-WithSource` parameter to `Publish-AzBicepModule` for publishing source with a module (currently experimental)
```
Comment (BethanyZhou) [inline src/Resources/Resources/ChangeLog.md:24]: @isra-fel, we may consider add a preview attribute for experimental parameter in our framework in future. Current preview attribute CmdletPreview is for cmdlet only.

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Resources/Resources/ChangeLog.md (hunk 4 above)
Comment (BethanyZhou) [inline src/Resources/Resources/ChangeLog.md:24]: Please insert change log between L21-L22, you may find conflicts then. Please resolve them.

Comment (isra-fel) [reply src/Resources/Resources/ChangeLog.md:26]: Hi @StephenWeatherford , how unstable is the -WithSource feature? Are you comfortable if we need it to follow the policy of breaking change? (breaking changes allowed only in major releases, major release being twice a year)

**Comment by dolauli** [comment]
@StephenWeatherford  Please fix the CI error and resolve the conflict.

File: src/Resources/Resources/ChangeLog.md (hunk 4 above)
Comment (isra-fel) [reply src/Resources/Resources/ChangeLog.md]: Got it. Let's make it this way.

**Comment by StephenWeatherford** [comment]
> @StephenWeatherford Please fix the CI error and resolve the conflict.

Done.

**Comment by dolauli** [comment]
@BethanyZhou Would you review the last change?

File: src/Resources/Resources/ChangeLog.md [hunk 5]
```diff
@@ -19,7 +19,8 @@
 -->
 
 ## Upcoming Release
-* Supported nullable Bicep parameters in Deployment cmdlets
+* Added `-WithSource` parameter to `Publish-AzBicepModule` for publishing source with a module (currently experimental)
+* Support nullable Bicep parameters in Deployment cmdlets
```
Comment (BethanyZhou) [inline src/Resources/Resources/ChangeLog.md:23]:
```suggestion
* Supported nullable Bicep parameters in Deployment cmdlets
```

**Review by BethanyZhou — APPROVED** [review]
Looks good to me
